test/test_indexing.py

```diff
@@ -649,6 +649,302 @@ def test_advancedindex_big(self, device):
         self.assertEqual(reference[[0, 123, 44488, 68807, 123343], ],
                          torch.tensor([0, 123, 44488, 68807, 123343], dtype=torch.int))
 
+    _advancedindex_input1 = [
+        [[0., 1.],
+         [2., 3.],
+         [4., 5.]],
+
+        [[6., 7.],
+         [8., 9.],
+         [10., 11.]],
+
+        [[12., 13.],
+         [14., 15.],
+         [16., 17.]],
+
+        [[18., 19.],
+         [20., 21.],
+         [22., 23.]]
+    ]
+
+    _advancedindex_input2 = [
+        [[24., 25.],
+         [26., 27.],
+         [28., 29.]],
+
+        [[30., 31.],
+         [32., 33.],
+         [34., 35.]],
+
+        [[36., 37.],
+         [38., 39.],
+         [40., 41.]],
+
+        [[42., 43.],
+         [44., 45.],
+         [46., 47.]]
+    ]
+
+    # one mask tensor (2d)
+    @dtypes(torch.float)
+    def test_advancedindex_dim_1m_2d(self, device, dtype):
+        t = self._advancedindex_input1
+
+        values = (
+            # tensor, mask, result
+            # case 0: everything
+            (t,
+
+             [[1., 1., 1.],
+              [1., 1., 1.],
+              [1., 1., 1.],
+              [1., 1., 1.]],
+
+             # 0   1   2   3   4   5    6    7    8    9    10   11
+             [0., 2., 4., 6., 8., 10., 12., 14., 16., 18., 20., 22.]),
+
+            # case 1: nothing
+            (t,
+
+             [[0., 0., 0.],
+              [0., 0., 0.],
+              [0., 0., 0.],
+              [0., 0., 0.]],
+
+             []),
+
+            # case 2: random
+            (t,
+
+             [[0., 0., 1.],   # 0  1  2
+              [0., 1., 0.],   # 3  4  5
+              [0., 0., 0.],   # 6  7  8
+              [0., 0., 1.]],  # 9 10 11
+
+             [4., 8., 22.]),
+        )
+
+        for tv, mv, rv in values:
+            t = torch.tensor(tv, device=device, dtype=dtype)
+            m = torch.tensor(mv, device=device, dtype=dtype) > 0
+            r = torch.tensor(rv, device=device, dtype=dtype)
```

Comment:
You wrap the data from above into a tensor anyway. Why not just use `ones`, `zeros` and `rand`/`randint` right away for all 3 cases?

Reply:
as with other cases, this is also later fed into numpy (right below the tensor/pytorch checks). i wanted to simplify this as much as possible (a unit test) rather than turning it into an integration test. or should i just go thru tensor/pytorch apis anyway, and later somehow convert into a format that can be fed into numpy? as you say, it'd reduce the amount of code, but it will also introduce lots of conversions and data mangling (reshaping etc). i have no strong opinion on this, just tell me what you'd prefer

Reply:
yeah, assuming this works when passing arguments to numpy, i think i'll try your suggestion despite this requiring other pytorch operations behaving properly. it'll make the code shorter and help with testing more things.

Reply:
You can call `Tensor.numpy()` to pass to numpy.